p2p/security/tls/transport_test.go

```diff
@@ -240,13 +346,72 @@ func TestHandshakeConnectionCancellations(t *testing.T) {
 	})
 }
 
+func TestHandshakeConnectionWithNextProtoCancellations(t *testing.T) {
```

Comment:
I don't think we need to test cancelations in combination with muxer selection.

Reply:
Removed.